### PR TITLE
E2E test commit

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: cypress/e2e/002-journeys/007-employee-mapping.cy.js [hunk 1]
```diff
@@ -47,19 +47,40 @@ describe('employee mapping view/create/update', () => {
...
-      const existingMappingValue = el.text()
+      existingMappingValue = el.text()
+
 
       cy.get('@employeeMappingRow').find('.mapping-table--form-field').click()
       // Select 1st option
-      cy.get('.mat-option').eq(0).click()
-
-      cy.get('@employeeMappingRow').find('.mapping-table--form-field').should('not.have.text', existingMappingValue)
+      cy.get('.mat-option').eq(el.index()+1).click()
```
Comment (ashwin1111) [inline cypress/e2e/002-journeys/007-employee-mapping.cy.js:62]: why is this changed?

File: cypress/e2e/002-journeys/007-employee-mapping.cy.js [hunk 2]
```diff
@@ -47,19 +47,40 @@ describe('employee mapping view/create/update', () => {
...
+    cy.get('.mat-option').eq(0).click()
+    cy.get('.mapping-table--form-field').eq(0).contains('Ashwin')
+  })
+
+  it('advanced search without data', () => {
+    cy.get('.mapping-header-section--card-content-text-header').contains('Unmapped Employees').click()
+    cy.get('.mapping-table--form-field').eq(0).contains('Select Vendor').click()
+    cy.get('.search-select--search-input').eq(1).type('ashwinlp')
+    cy.get('.mat-option').eq(0).contains('Searching...')
+    cy.wait(1000)
```
Comment (ashwin1111) [inline cypress/e2e/002-journeys/007-employee-mapping.cy.js:81]: wait till API call is completed, add an alias and wait till the request is completed
example - `cy.wait('@getDestinationAttributes').its('response.statusCode').should('equal', 200)`

File: cypress/e2e/002-journeys/008-category-mapping.cy.js [hunk 3]
```diff
@@ -47,19 +47,41 @@ describe('category mapping view/create/update', () => {
...
     cy.get('@categoryMappingRow').find('.mapping-table--form-field').then((el) => {
-      const existingMappingValue = el.text()
+      existingMappingValue = el.text()
 
       cy.get('@categoryMappingRow').find('.mapping-table--form-field').click()
       // Select 1st option
-      cy.get('.mat-option').eq(0).click()
-
-      cy.get('@categoryMappingRow').find('.mapping-table--form-field').should('not.have.text', existingMappingValue)
+      cy.get('.mat-option').eq(el.index()+1).click()
```
Comment (ashwin1111) [inline cypress/e2e/002-journeys/008-category-mapping.cy.js:61]: why is this changed?

File: cypress/e2e/002-journeys/008-category-mapping.cy.js [hunk 4]
```diff
@@ -47,19 +47,41 @@ describe('category mapping view/create/update', () => {
...
     })
+    cy.get('@categoryMappingRow').find('.mapping-table--form-field').should('not.have.text', existingMappingValue)
+  })
+
+  it('advanced search', () => {
+    cy.get('.mapping-header-section--card-content-text-header').contains('Unmapped Categories').click()
+    cy.get('.mapping-table--form-field').eq(0).contains('Select Account').click()
+    cy.get('.search-select--search-input').eq(1).type('food')
+    cy.get('.mat-option').eq(0).contains('Searching...')
+    cy.wait(1000)
```
Comment (ashwin1111) [inline cypress/e2e/002-journeys/008-category-mapping.cy.js:71]: same

File: cypress/e2e/002-journeys/008-category-mapping.cy.js [hunk 5]
```diff
@@ -47,19 +47,41 @@ describe('category mapping view/create/update', () => {
...
+    cy.get('.mat-option').eq(0).click()
+    cy.get('.mapping-table--form-field').eq(0).contains('Food')
+  })
+
+  it('advanced search without data', () => {
+    cy.get('.mapping-header-section--card-content-text-header').contains('Unmapped Categories').click()
+    cy.get('.mapping-table--form-field').eq(0).contains('Select Account').click()
+    cy.get('.search-select--search-input').eq(1).type('fooopd')
+    cy.get('.mat-option').eq(0).contains('Searching...')
+    cy.wait(1000)
```
Comment (ashwin1111) [inline cypress/e2e/002-journeys/008-category-mapping.cy.js:82]: same

File: cypress/e2e/002-journeys/007-employee-mapping.cy.js [hunk 6]
```diff
@@ -47,19 +47,40 @@ describe('employee mapping view/create/update', () => {
   })
 
   it('update employee mappings', () => {
+    let existingMappingValue = ''
```
Comment (ashwin1111) [inline cypress/e2e/002-journeys/007-employee-mapping.cy.js:50]: we define a mappingLimit value, make it 10 for all the mapping pages. Previously it used to return all values.

File: cypress/e2e/002-journeys/002-update-configuration.cy.js [hunk 7]
```diff
@@ -298,7 +298,7 @@ describe('update configuration', () => {
     cy.get('@categoryMappingRow').find('.mat-column-fyle').contains('Food')
 
     cy.get('@categoryMappingRow').find('.mapping-table--form-field').click()
-    cy.get('.mat-option').contains('Opening Balance Equity').click()
+    cy.get('.mat-option').contains('Bank Charges').click()
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply cypress/e2e/002-journeys/002-update-configuration.cy.js:301]: it was failing as we limit the result, I will try to do a advanced search and select 'Opening Balance Equity'.

File: cypress/e2e/002-journeys/005-resolve-qbo-error.cy.js [hunk 8]
```diff
@@ -64,7 +64,7 @@ describe('resolve qbo error', () => {
     cy.get('@categoryMappingRow').find('.mat-column-fyle').contains('Food')
 
     cy.get('@categoryMappingRow').find('.mapping-table--form-field').click()
-    cy.get('.mat-option').contains('Food').click()
+    cy.get('.mat-option').contains('Automobile').click()
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply cypress/e2e/002-journeys/005-resolve-qbo-error.cy.js:67]: yes there is no error, I will right them

[the remark this answers is not in the document]
Comment (ashwin1111) [reply cypress/e2e/002-journeys/005-resolve-qbo-error.cy.js:67]: thanks

File: cypress/e2e/002-journeys/002-update-configuration.cy.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (ashwin1111) [reply cypress/e2e/002-journeys/002-update-configuration.cy.js:301]: thanks

File: cypress/e2e/002-journeys/008-category-mapping.cy.js (hunk 3 above)
Comment (DhaaraniCIT) [reply cypress/e2e/002-journeys/008-category-mapping.cy.js:61]: this is because, some times the pervious value is taken as the current value, when it is inside the "then", so this test was failing, so i keept it outside the "then" block.
And by doing index+1, the next value is selected

Comment (ashwin1111) [reply cypress/e2e/002-journeys/008-category-mapping.cy.js:61]: we never select the 1st option, we will have a random option selected from 2nd till end. That's why we change it to first option, so that the value selected before and after will never be the same

File: cypress/e2e/002-journeys/002-update-configuration.cy.js [hunk 9]
```diff
@@ -298,6 +298,8 @@ describe('update configuration', () => {
     cy.get('@categoryMappingRow').find('.mat-column-fyle').contains('Food')
 
     cy.get('@categoryMappingRow').find('.mapping-table--form-field').click()
+    cy.get('.search-select--search-input').eq(1).type('Opening Balance Equity')
```
Comment (ashwin1111) [inline cypress/e2e/002-journeys/002-update-configuration.cy.js:301]: this is already there in 303 line no?

File: cypress/e2e/002-journeys/002-update-configuration.cy.js [hunk 10]
```diff
@@ -308,7 +308,6 @@ describe('update configuration', () => {
     cy.url().should('include', '/workspaces/main/mapping/project')
 
     cy.get('.mapping-filter--filter-alphabet-list').contains('F').click()
-    cy.wait('@getDestinationAttributes').its('response.statusCode').should('equal', 200)
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply cypress/e2e/002-journeys/002-update-configuration.cy.js:311]: @DhaaraniCIT why is this marked as resolved?

File: cypress/e2e/002-journeys/002-update-configuration.cy.js [hunk 11]
```diff
@@ -298,6 +298,8 @@ describe('update configuration', () => {
     cy.get('@categoryMappingRow').find('.mat-column-fyle').contains('Food')
 
     cy.get('@categoryMappingRow').find('.mapping-table--form-field').click()
+    cy.get('.search-select--search-input').eq(1).type('Opening Balance Equity')
+    cy.wait(1000)
```
Comment (ashwin1111) [inline cypress/e2e/002-journeys/002-update-configuration.cy.js:302]: We shouldn't use wait with time. Check https://github.com/fylein/fyle-qbo-app-2/pull/220#discussion_r1005266531

File: cypress/e2e/002-journeys/005-resolve-qbo-error.cy.js [hunk 12]
```diff
@@ -64,6 +64,8 @@ describe('resolve qbo error', () => {
     cy.get('@categoryMappingRow').find('.mat-column-fyle').contains('Food')
 
     cy.get('@categoryMappingRow').find('.mapping-table--form-field').click()
+    cy.get('.search-select--search-input').eq(1).type('food')
+    cy.wait(1000)
```
Comment (ashwin1111) [inline cypress/e2e/002-journeys/005-resolve-qbo-error.cy.js:68]: We shouldn't use wait with time. Check https://github.com/fylein/fyle-qbo-app-2/pull/220#discussion_r1005266531

File: cypress/e2e/002-journeys/007-employee-mapping.cy.js [hunk 13]
```diff
@@ -47,19 +47,43 @@ describe('employee mapping view/create/update', () => {
...
-      cy.get('@employeeMappingRow').find('.mapping-table--form-field').should('not.have.text', existingMappingValue)
     })
+    cy.get('@employeeMappingRow').find('.mapping-table--form-field').should('not.have.text', existingMappingValue)
+  })
+
+  it('advanced search', () => {
+    cy.get('.mapping-header-section--card-content-text-header').contains('Unmapped Employees').click()
+    cy.get('.mapping-table--form-field').eq(0).contains('Select Vendor').click()
+    cy.get('.search-select--search-input').eq(1).type('ashwin')
+    cy.wait(1000)
```
Comment (ashwin1111) [inline cypress/e2e/002-journeys/007-employee-mapping.cy.js:71]: We shouldn't use wait with time. Check https://github.com/fylein/fyle-qbo-app-2/pull/220#discussion_r1005266531

File: cypress/e2e/002-journeys/008-category-mapping.cy.js (hunk 4 above)
Comment (ashwin1111) [reply cypress/e2e/002-journeys/008-category-mapping.cy.js:71]: We shouldn't use wait with time. Check https://github.com/fylein/fyle-qbo-app-2/pull/220#discussion_r1005266531

File: cypress/e2e/002-journeys/008-category-mapping.cy.js (hunk 5 above)
Comment (ashwin1111) [reply cypress/e2e/002-journeys/008-category-mapping.cy.js:83]: We shouldn't use wait with time. Check https://github.com/fylein/fyle-qbo-app-2/pull/220#discussion_r1005266531

File: cypress/support/commands.ts [hunk 14]
```diff
@@ -79,6 +79,10 @@ Cypress.Commands.add('setupHttpListeners', () => {
   setupInterceptor('GET', '/fyle/expense_fields', 'getFyleExpenseFields')
 
   setupInterceptor('GET', '/mappings/employee_attributes/', 'getEmployeeMappings')
+
+  setupInterceptor('GET', '/qbo/vendors/', 'getQBOVendors')
+
+  setupInterceptor('GET', '/qbo/mapping_options/', 'getQBOCategories')
```
Comment (ashwin1111) [inline cypress/support/commands.ts:85]: name it getQBOOptions / getMappingOptions. This is a generic endpoint and not restricted to category

File: cypress/e2e/002-journeys/002-update-configuration.cy.js (hunk 9 above)
Comment (DhaaraniCIT) [reply cypress/e2e/002-journeys/002-update-configuration.cy.js:301]: line 303 is selecting process, this is just typing

File: cypress/e2e/002-journeys/002-update-configuration.cy.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply cypress/e2e/002-journeys/002-update-configuration.cy.js:311]: The '@getDestinationAttribute' api call is different and the call we do is different now as we have "mapping_option" so i have created a new alise and placed it in this place as "cy.wait('@getQBOCategories').its('response.statusCode').should('equal', 200)"

File: cypress/e2e/002-journeys/002-update-configuration.cy.js (hunk 9 above)
Comment (ashwin1111) [reply cypress/e2e/002-journeys/002-update-configuration.cy.js:301]: got it

File: cypress/e2e/002-journeys/002-update-configuration.cy.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (ashwin1111) [reply cypress/e2e/002-journeys/002-update-configuration.cy.js:311]: got it

**Review by ashwin1111 — APPROVED** [review]
